samples/matter/common/src/diagnostic/diagnostic_logs_crash.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "diagnostic_logs_crash.h"
+
+#include <cstdlib>
+#include <zephyr/kernel.h>
+#include <zephyr/retention/retention.h>
+
+using namespace Nrf;
+
+#define Verify(expr, value)                                                                                            \
+	do {                                                                                                           \
+		if (!(expr)) {                                                                                         \
+			return (value);                                                                                \
+		}                                                                                                      \
+	} while (false)
+
+size_t CrashData::CalculateSize()
+{
+	bool unusedOut;
+	return ProcessConversionToLog(nullptr, 0, unusedOut);
+}
+
+size_t CrashData::ProcessConversionToLog(char *outBuffer, size_t bufferSize, bool &end)
+{
+	/* Allow to enter outBuffer as nullptr to calculate actual size.
+	if buffer is available, its size cannot be 0 */
+	if (outBuffer && bufferSize == 0) {
+		return 0;
+	}
+	mOutBufferSize = bufferSize;
+	mOutBuffer = outBuffer;
+	mOffset = 0;
+	mLine = 0;
+	end = false;
+
+	/* Process all stages one by one until reaching the buffer full or the end of the logs. */
+	Verify(BasicDump(), mOffset);
+	Verify(ReasonDump(), mOffset);
+	end = true;
+
+	return mOffset;
+}
+
+bool CrashData::BasicDump()
+{
+	Verify(Collect("Faulting instruction address (r15/pc): 0x%08x\n", mDescription.Esf.basic.pc), false);
```

Comment:
Well, I'm not the author of this one :smile: But considering it's a crash log, so something went wrong anyway, maybe it's a good idea to at least save a part of log instead of dropping it and having nothing saved?

Reply:
It is better to return false without converting the whole dump. If one of the Verify returns false, then the mechanism is stopped and the dump is not saved. So we avoid dumping the partial log, no matter if we return false at the end, or beginning.